src/sortStudents.ts

```diff
@@ -1,7 +1,92 @@
-// describe Student type
-// create and export SortType enum
-// create SortOrder type
+/* eslint-disable implicit-arrow-linebreak */
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+interface Student {
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
+
+export enum SortType {
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
+}
+
+type SortOrder = 'asc' | 'desc';
+
+const getAvgGrade = (student: Student): number => {
+  const total = student.grades.reduce(
+    (sum: number, grade: number) => sum + grade, 0,
+  );
+
+  return total / student.grades.length;
+};
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copy: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc' || '') {
+        return copy.sort((a: Student, b: Student) =>
+          a[sortBy].localeCompare(b[sortBy]));
+      }
+
+      if (order === 'desc') {
+        return copy.sort((a: Student, b: Student) =>
+          b[sortBy].localeCompare(a[sortBy]));
+      }
```

Comment:
with ternary your code look much better

```suggestion
        return order === 'asc'
          ? copy.sort((a: Student, b: Student) =>
            a[sortBy].localeCompare(b[sortBy]))
          : copy.sort((a: Student, b: Student) =>
            b[sortBy].localeCompare(a[sortBy]))
```